### PR TITLE
Update with bokeh>=2.4 and ipykernel>=6 compatibility

### PR DESCRIPTION
**Pull request by philippjfr**
@mattpap Please review, merge and release.

Fixes https://github.com/bokeh/ipywidgets_bokeh/issues/35
Fixes https://github.com/bokeh/ipywidgets_bokeh/issues/30

### CODE REVIEW

File: ipywidgets_bokeh/package.json [hunk 1]
```diff
@@ -14,15 +14,15 @@
     "url": "https://github.com/bokeh/ipywidgets_bokeh.git"
   },
   "dependencies": {
-    "@bokeh/bokehjs": "~2.0.0",
+    "@bokeh/bokehjs": "^2.4.0",
```
Comment (mattpap) [inline ipywidgets_bokeh/package.json:17]: Is this change intentional? Given a version `x.y.z`, `^` operator allows changes at `y`, whereas `~` operator allows changes at `z`. Given changes to `install_requires`, I presume `~` should be kept.

Comment (philippjfr) [reply ipywidgets_bokeh/package.json:17]: You're right, thanks!

File: ipywidgets_bokeh/kernel.py [hunk 2]
```diff
@@ -39,10 +48,13 @@ def send(self, stream, msg_type, content=None, parent=None, ident=None, buffers=
...
-        from bokeh.document.events import MessageSentEvent
-
         doc = self.document
-        doc.on_message("ipywidgets_bokeh", self.receive)
+
+        # Ensure document message handler is only added once
+        try:
+            doc.remove_on_message("ipywidgets_bokeh", self.receive)
+        finally:
+            doc.on_message("ipywidgets_bokeh", self.receive)
```
Comment (philippjfr) [inline ipywidgets_bokeh/kernel.py:57]: Don't love this of course but previously we were attaching this callback multiple times.

Comment (philippjfr) [reply ipywidgets_bokeh/kernel.py:58]: Was `on_message` not idempotent before bokeh 2.4?
